### PR TITLE
Pro 6136 jsx tutorial

### PR DESCRIPTION
**Pull request by BoDonkey**
## Summary

*Summarize the changes briefly, including which issue/ticket this resolves. If it closes an existing Github issue, include "Closes #[issue number]"*
This PR adds a tutorial for the creation of a weather widget using JSX and React. It focuses on the Webpack alterations needed, as opposed to the React code. Closes PRO-6136

## What are the specific steps to test this change?

*For example:*
> 1. Run the website and log in as an admin
> 2. Open a piece manager modal and select several pieces
> 3. Click the "Archive" button on the top left of the manager and confirm that it should proceed
> 4. Check that all pieces have been archived properly

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [X] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

**Review by boutell — CHANGES_REQUESTED** [review]

File: docs/.vitepress/sidebarTutorials.js [hunk 1]
```diff
@@ -66,12 +66,17 @@ const sidebarTutorials = [
     collapsed: false,
     items: [
       {
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/.vitepress/sidebarTutorials.js]: Dagnabbit!

File: docs/tutorials/using-jsx-in-apostrophe.md [hunk 2]
```diff
@@ -0,0 +1,673 @@
...
+  fields: {
+    // schema field code
+  },
+  webpack: {
+    // webpack configuration code
+  },
+  apiRoutes(self) {
+    return {
+      get: {
+        '/fetch-weather': async function (req, res) {
```
Comment (boutell) [inline docs/tutorials/using-jsx-in-apostrophe.md:355]: This isn't a public-facing, address bar URL, so I would do:

```
async fetchWeather(...) { ... }
```

And on the front end I'd access:

```
/api/v1/react-weather-widget/fetch-weather
```

This automatically namespaces the endpoint for you and allows you to use a more natural method name, it is automatically converted to kebab case.

**Review by boutell — CHANGES_REQUESTED** [review]

File: docs/tutorials/using-jsx-in-apostrophe.md [hunk 3]
```diff
@@ -297,7 +297,7 @@ function App({ defaultCity }) {
 
   const fetchWeather = async (cityName) => {
     try {
-      const response = await fetch(`/fetch-weather?city=${cityName}`);
+      const response = await fetch(`/api/v1/react-weather-widget/fetch-weather?city=${cityName}`);
```
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/tutorials/using-jsx-in-apostrophe.md:300]: Changed and tested.

File: docs/tutorials/using-jsx-in-apostrophe.md [hunk 4]
```diff
@@ -352,7 +352,7 @@ module.exports = {
   apiRoutes(self) {
     return {
       get: {
-        '/fetch-weather': async function (req, res) {
+        fetchWeather: async function (req, res) {
```
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/tutorials/using-jsx-in-apostrophe.md:355]: Changed.

File: docs/tutorials/using-jsx-in-apostrophe.md [hunk 5]
```diff
@@ -376,7 +376,9 @@ module.exports = {
...
+
+The remainder of this code should be fairly self-explanatory. We are getting the `city` value from the request object and the API key from an `.env` file in our project. Note that at the top of the file we are using `require('dotenv').config();` to expose this. From Node.js version `20.0.0` and up, this is supported by passing the `--env-file` flag specifying a file when starting your app and doesn't require the `dotenv` dependency. You can also do away with this line if you are passing your API key directly on the command line when starting your project => `OPENWEATERMAP_API_KEY=XXXXXX npm run dev`.
```
[the remark this answers is not in the document]
Comment (BoDonkey) [reply docs/tutorials/using-jsx-in-apostrophe.md:381]: Fixed and env file reference removed.

File: docs/tutorials/using-jsx-in-apostrophe.md [hunk 6]
```diff
@@ -173,17 +177,18 @@ import App from './jsx-components/App.jsx';
 export default () => {
   apos.util.widgetPlayers.reactWeather = {
     selector: '[data-react-weather-widget]',
-    player: function(el) {
+    player: function (el) {
```
Comment (boutell) [inline docs/tutorials/using-jsx-in-apostrophe.md:180]: Method syntax is nicer here (an actual class is not required):

```
player(el) { ... }
```

File: docs/tutorials/using-jsx-in-apostrophe.md [hunk 7]
```diff
@@ -195,6 +200,7 @@ At the top of this file we are importing both `react` and the `createRoot` funct
 The remainder of this file is a standard widget player. This player is attached to the `[data-react-weather-widget]` attribute that we will need to add to the widget Nunjucks template. Within that element, it selects an element with an id of `react-weather-root` to create the root for our React component. We are also passing a prop we are getting from the `data-default-city` attribute on our `rootElement`. We will need to add this root element and the data into the data attribute that we will be getting from a schema field.
```
Comment (boutell) [inline docs/tutorials/using-jsx-in-apostrophe.md:200]: Link to widget player docs

Comment (boutell) [reply docs/tutorials/using-jsx-in-apostrophe.md:200]: Something about the last sentence seems confusing but it's hard to put my finger on it

File: docs/tutorials/using-jsx-in-apostrophe.md [hunk 8]
```diff
@@ -297,11 +305,16 @@ function App({ defaultCity }) {
 
   const fetchWeather = async (cityName) => {
     try {
-      const response = await fetch(`/api/v1/react-weather-widget/fetch-weather?city=${cityName}`);
+      const response = await fetch(
+        '/api/v1/react-weather-widget/fetch-weather?' +
+          new URLSearchParams({
```
Comment (boutell) [inline docs/tutorials/using-jsx-in-apostrophe.md:310]: Nice

File: docs/tutorials/using-jsx-in-apostrophe.md [hunk 9]
```diff
@@ -312,14 +325,18 @@ function App({ defaultCity }) {
   return (
     <Container>
       <AppLabel>React Weather App</AppLabel>
-      <CityComponent updateCity={updateCity} fetchWeather={handleFetchWeather} />
+      <CityComponent
+        updateCity={updateCity}
```
Comment (boutell) [inline docs/tutorials/using-jsx-in-apostrophe.md:329]: camel case is... standard in react... react is weird

File: docs/tutorials/using-jsx-in-apostrophe.md [hunk 10]
```diff
@@ -441,15 +467,16 @@ const CityComponent = (props) => {
       <SearchBox onSubmit={fetchWeather}>
         <input
           onChange={(e) => updateCity(e.target.value)}
-          placeholder="City"
+          placeholder='City'
         />
-        <button type={"submit"}>Search</button>
+        <button type={'submit'}>Search</button>
```
Comment (boutell) [inline docs/tutorials/using-jsx-in-apostrophe.md:472]: It sounds like the curlies aren't adding any value here

**Review by boutell — CHANGES_REQUESTED** [review]

File: docs/tutorials/using-jsx-in-apostrophe.md (hunk 6 above)
Comment (BoDonkey) [reply docs/tutorials/using-jsx-in-apostrophe.md:180]: Changed.

File: docs/tutorials/using-jsx-in-apostrophe.md (hunk 7 above)
Comment (BoDonkey) [reply docs/tutorials/using-jsx-in-apostrophe.md:200]: Added and re-phrased.

File: docs/tutorials/using-jsx-in-apostrophe.md (hunk 9 above)
Comment (BoDonkey) [reply docs/tutorials/using-jsx-in-apostrophe.md:329]: React liberally mixes camel, pascal, kebab, and screaming snake.

File: docs/tutorials/using-jsx-in-apostrophe.md (hunk 10 above)
Comment (BoDonkey) [reply docs/tutorials/using-jsx-in-apostrophe.md:472]: I think I had refactored that from using a variable and forgot to remove the curlies. Thanks.

**Review by boutell — APPROVED** [review]

**Pull request merged**